aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1313,65 +1298,44 @@ Tensor &frobenius_norm_out(
       "Expected at most 2 dimensions, but got ",
       dim.size(),
       " dimensions instead.");
+  Tensor result_;
   if (dim.size() == 1 || dim.size() == 0) {
-    return at::norm_out(result, self, 2, dim, keepdim, self.scalar_type());
-  }
-  auto dim_ = dim.vec();
-  maybe_wrap_dims(dim_, self.dim());
-  TORCH_CHECK(dim_[0] != dim_[1], "Expected dims to be different, got ", dim, " instead");
-  if (self.is_complex()){
-    return at::sqrt_out(result, at::sum(at::real(self.conj() * self), dim_, keepdim));
+    result_ = at::norm(self, 2, dim, keepdim);
```

Comment:
Can this use `norm_out` and, below, `sqrt_out` to avoid copying into `result`?

Reply:
Sure, I'll change it to `norm_out`. However, `sqrt_out` is evidently not differentiable yet:
```
>>> a = torch.randn(10, requires_grad=True)
>>> b = torch.empty(10)
tensor([7.3212e-26, 3.0963e-41, 5.1732e+09, 4.5619e-41, 8.9683e-44, 0.0000e+00,
        1.1210e-43, 0.0000e+00, 7.3956e-26, 3.0963e-41])
>>> torch.sqrt(a, out=b)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: sqrt(): functions with out=... arguments don't support automatic differentiation, but one of the arguments requires grad.
```

Reply:
Actually, looks like there's an issue with using `at::norm_out`. I'm not really sure why that is.

```
$ python test/test_autograd.py TestAutogradDeviceTypeCPU.test_norm_default_cpu
...
  File "/opt/conda/lib/python3.8/site-packages/torch/functional.py", line 1256, in norm
    return _VF.frobenius_norm(input, dim=(), keepdim=keepdim)  # type: ignore
RuntimeError: norm(): functions with out=... arguments don't support automatic differentiation, but one of the arguments requires grad.
```

Reply:
I guess this is a legitimately missing feature. The error is thrown from `throw_error_out_requires_grad()` in generated code in `torch/csrc/autograd/generated/VariableTypeEverything.cpp`:
```
Tensor & norm_out_out(Tensor & out, const Tensor & self, c10::optional<Scalar> p, IntArrayRef dim, bool keepdim) {
  auto& out_ = unpack(out, "out", 0);
  auto& self_ = unpack(self, "self", 1);
  std::shared_ptr<Node> grad_fn;
  if (compute_requires_grad( self )) {
    throw_error_out_requires_grad("norm");
  }
  if (compute_requires_grad( out )) {
    throw_error_out_requires_grad("norm");
  }
  ...
```
So I think I'll have to revert back to the resize and copy approach.

Reply:
Darn. Let's add a comment to this function noting we tried that and ran into this autograd issue. That should avoid someone trying the same thing in the future (unless they intend to spend some time addressing the issues).